### PR TITLE
fix(oracle): clamp upscaled price to uint32-safe bounds

### PR DESCRIPTION
**Pull request by 0xCardiE**
## Summary
- Cast to `uint64` before upscaling in `setPrice` to avoid overflow/truncation when shifting.
- Add `MAX_CURRENT_PRICE_UPSCALED` and `_clampPriceUpscaled` so `currentPrice()` cannot disagree with `currentPriceUpScaled` or under-report vs `minimumPrice()`.

## Test plan
- [ ] Run existing PriceOracle unit tests
- [ ] Verify `setPrice` and round adjustment clamp to `[minimumPriceUpscaled, MAX_CURRENT_PRICE_UPSCALED]`
- [ ] Confirm `currentPrice()` matches `currentPriceUpScaled >> 10` at boundary values

### CODE REVIEW

File: src/PriceOracle.sol [hunk 1]
```diff
@@ -41,6 +41,11 @@ contract PriceOracle is AccessControl {
     // The length of a round in blocks.
     uint8 private constant ROUND_LENGTH = 152;
 
+    /// @dev Upper bound for upscaled price so `(currentPriceUpScaled >> 10)` fits in `uint32`.
+    /// Without this, `currentPrice()`'s `uint32(... >> 10)` truncates and can disagree with
+    /// `currentPriceUpScaled` and under-report vs `minimumPrice()`.
+    uint64 public constant MAX_CURRENT_PRICE_UPSCALED = uint64(uint256(type(uint32).max) << 10);
```
Comment (nugaon) [inline src/PriceOracle.sol]: I would have just cast once into uint64 but it is fine

File: src/PriceOracle.sol [hunk 2]
```diff
@@ -182,6 +179,18 @@ contract PriceOracle is AccessControl {
     //            STATE READING           //
     ////////////////////////////////////////
 
+    /// @notice Clamp upscaled price to [minimumPriceUpscaled, MAX_CURRENT_PRICE_UPSCALED].
+    function _clampPriceUpscaled(uint64 priceUpScaled) private view returns (uint64) {
+        uint64 minU = uint64(minimumPriceUpscaled);
```
Comment (nugaon) [inline src/PriceOracle.sol]: this `minimumPriceUpscaled` should be a constant to save gas. `uint64 public constant MINIMUM_PRICE_UPSCALED = 24000 << 10;`

**Review by nugaon — APPROVED** [review]
